src/components/checkbox/index.tsx

```diff
@@ -1,56 +0,0 @@
-import { useEffect, useState } from "react";
-import axios from "axios";
-import { Checkbox } from "antd";
-import "../../App.css";
-interface CheckboxProps {
-  checkboxNumber: number;
-}
-
-const CustomCheckbox = ({ checkboxNumber }: CheckboxProps) => {
-  const [letters, setLetters] = useState<string[]>([]);
-  const [checkvalue, setCheckValue] = useState<boolean>(false);
-  const url =
-    "https://navirego-interview-mc3narrsb-volodymyr-matselyukh.vercel.app/api/letters/";
-  useEffect(() => {
-    const handleFetch = async () => {
-      const res = await axios.get(`${url}${checkboxNumber}`);
-      const currentLetters = [...letters];
-      if (currentLetters.length >= 30) {
-        currentLetters.shift();
-      }
-      const result = res.data.letter;
-      setLetters(() => [...currentLetters, result]);
-    };
-    let intervalId: NodeJS.Timeout;
-    if (checkvalue) {
-      intervalId = setInterval(handleFetch, 2000);
-    }
-
```

Comment:
A single useEffect should not store a big logic like this. We don't cramp up things just because it can handle.
An ideal way of thinking is imagine that you have to write test cases for the application. If each function is independent then you will not have to worry about big functions. Small functions which are unit testable is a better choice.